### ISSUE TITLE
Virutalenv support for building bottleneck

### ISSUE THREAD

**Issue by Zeitsperre** [pydata/bottleneck#396]
I have been trying to get Bottleneck to build from master using Python3.10 in the GitHub Actions environment. After finding the deadsnakes python action for installing the python-dev headers in Ubuntu 20.04 (https://github.com/deadsnakes/action), I was still having issues with the builds failing (despite everything working fine locally).

The fix to this issue raised here was actually solved by my colleague @aulemahal in another issue. I am opening this issue for context, with a PR incoming.

Explanation:
----
I believe I have found the source of the error, at least in my configuration. It is due to the specific of virtual environment creation with `python -m venv`. 

When creating a virtual environment with `venv` into `${venv}`, a `${venv}/include` folder is created and `sys.prefix == ${venv}`. However, in opposition to `virtualenv`, the include folder **is empty**, it exists solely for files created by packages installed _within_ the environment.  Source : [PEP 405](https://www.python.org/dev/peps/pep-0405/#id22).

The include folders needed by bottleneck are still accessible, but their path has to be constructed relative to `sys.base_prefix` instead, to get the system files. Changing: 
https://github.com/pydata/bottleneck/blob/f2bc792e540908d165143c4930d63d1e7ab806d3/bottleneck/src/bn_config.py#L30
to
```python3
    for prefix in [sys.prefix, sys.exec_prefix, sys.base_prefix]:
```
fixes the problem.

Of course, I have `pythonX.X-dev` installed (on ubuntu) for this to work.

EDIT: I have python 3.7.9.

_Originally posted by @aulemahal in https://github.com/pydata/bottleneck/issues/348#issuecomment-776165418_

**Comment by chaseleif** [comment]
Thanks, this worked for me too. I was getting stuck on bottleneck not installing, this solved my issue.
Python 3.10.2, pip 22.0.3, Arch Linux 5.16.10

**Comment by Zeitsperre** [comment]
This appears to be fixed in the latest Bottleneck (v1.3.4). Thanks!
